doc/CHANNELING-API.txt

```diff
@@ -1035,6 +1044,58 @@ File sharing data channel protocol
         message was received.
 
 
+TURN cedentials and URNs document
+
+  TURN example data
+
+    {
+        "username": "turn-username",
+        "password": "turn-password",
+        "ttl": 3600
+        "urls": [
+            "turn:213.203.211.154:3478?transport=udp",
+            "turn:213.203.211.154:3479?transport=tcp",
+            "turns:213.203.211.154:443?transport=tcp"
+        ]
+    }
+
+  TURN data is provided as a mapping of key value pairs like `urls`, `password`
+  and `username`. See http://tools.ietf.org/html/draft-uberti-behave-turn-rest-00
+  and the TURN REST API section in https://github.com/coturn/coturn for details.
+
+  In addition, if the service is configured to use a TURN service to provide
+  credentials for TURN, it can can contain `geo_uri` and `servers` as returned
+  by the TURN service client. See https://github.com/strukturag/spreed-turnservicecli
+  for documentation.
+
+  TURN example data with TURN service
+
+    {
+      "geo_uri": "https://turnservice.spreed.me/api/v1/turn/geo",
+      "servers": [
+        {
+          "prio": 10,
+          "urns": [
+            "turn:213.203.211.154:3478?transport=udp",
+            "turn:213.203.211.154:3479?transport=tcp",
+            "turns:213.203.211.154:443?transport=tcp"
+          ],
+          "id": "zone-id"
+        }
+      ],
+      "ttl": 3600,
+      "username": "turn-username",
+      "password": "turn-password"
+    }
+
+  The TURN credentials data does expire after a while and needs to be refreshed
+  before the `ttl` is reached. The `ttl` is the number of seconds from now until
+  the data expires. To refresh the TURN data, a `Self` document with empty data
+  should be sent.
+
+  All values in the TURN data except the `ttl` are optional.
```

Comment:
Does that make sense? I think you should either get a `servers` list or a `urls` list.